### PR TITLE
net/lorawan/hdr: add lorawan header helpers

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR adds helpers for parsing and building LoRaWAN headers.
It's a part of #11022 and I put it here to make the review easier.

Also, this header is not GNRC LoRaWAN specific but can be used by any stack.

### Testing procedure
The commits were taken from #11022. So, they can be seen in action there :)
Alternatively, check the [LoRaWAN 1.0.3 specification](https://lora-alliance.org/sites/default/files/2018-07/lorawan1.0.3.pdf), chapter 4 (MAC Message format).

Also, test the documentation with `make doc`.

### Issues/PRs references
Part of #11022

### CODE REVIEW

File: sys/include/net/lorawan/hdr.h [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+ */
+static inline bool lorawan_hdr_get_frame_pending(lorawan_hdr_t *hdr)
+{
+    return (hdr->fctrl & 0x10);
+}
+
+/**
+ * @brief Set LoRaWAN header FOpts length
+ *
+ * @param hdr LoRaWAN header
```
Comment (smlng) [inline sys/include/net/lorawan/hdr.h]: missing `[in]` here, next and below

File: sys/include/net/lorawan/hdr.h [hunk 2]
```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (C) 2016 Fundación Inria Chile
```
Comment (smlng) [inline sys/include/net/lorawan/hdr.h]: this seems a bit outdated?

File: sys/include/net/lorawan/hdr.h [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+/**
+ * @defgroup    net_lorawan_hdr    LoRaWAN header
+ * @ingroup     net_lorawan
+ * @brief       LoRaWAN header types and helper functions
+ * @{
+ *
+ * @file
+ * @brief   LoRaWAN header type and helper function definitions
+ *
+ * @author  José Ignacio Alamos <jose.alamos@inria.cl>
```
Comment (smlng) [inline sys/include/net/lorawan/hdr.h]: does the mail still work :smile:

File: sys/include/net/lorawan/hdr.h [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+#include "net/loramac.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Data type to represent a LoRaWAN packet header
+ *
+ * @details This definition includes MHDR and FHDR in the same structure.
```
Comment (smlng) [inline sys/include/net/lorawan/hdr.h]: I don't think that `@details` is needed (here and below). At least it is hardly used in the RIOT docu else where

File: sys/include/net/lorawan/hdr.h [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+/**
+ * @brief Set LoRaWAN header MType
+ *
+ * @param[in] hdr LoRaWAN hdr
+ * @param[in] mtype desired MType
+ */
+static inline void lorawan_hdr_set_mtype(lorawan_hdr_t *hdr, uint8_t mtype)
+{
+    hdr->mt_maj &= 0x1f;
```
Comment (aabadie) [inline sys/include/net/lorawan/hdr.h]: I think those hard coded values should be provided as defines. What about `LORAWAN_HDR_MTYPE_MASK`, `LORAWAN_HDR_MTYPE_POS` or something ? The same could apply with other fields in the header.

File: sys/include/net/lorawan/hdr.h (hunk 3 above)
Comment (jia200x) [reply sys/include/net/lorawan/hdr.h]: Dunno, will check :)

File: sys/include/net/lorawan/hdr.h (hunk 2 above)
Comment (jia200x) [reply sys/include/net/lorawan/hdr.h]: In theory I produced this file while I was working there. So maybe I should keep it. I will add the HAW Hamburg copyright though

File: sys/include/net/lorawan/hdr.h (hunk 5 above)
Comment (jia200x) [reply sys/include/net/lorawan/hdr.h]: agreed

Comment (jia200x) [reply sys/include/net/lorawan/hdr.h]: done

File: sys/include/net/lorawan/hdr.h (hunk 4 above)
Comment (jia200x) [reply sys/include/net/lorawan/hdr.h]: removed

File: sys/include/net/lorawan/hdr.h (hunk 2 above)
Comment (jia200x) [reply sys/include/net/lorawan/hdr.h]: resolved? :)

**Comment by jia200x** [comment]
@aabadie @smlng I think I've addressed all coments

**Comment by jia200x** [comment]
ping @aabadie @smlng

**Comment by smlng** [comment]
please squash

**Comment by jia200x** [comment]
@smlng done

**Comment by smlng** [comment]
Ci is not happy with your documentation 😢

**Comment by jia200x** [comment]
I had to add the `net_lorawan` group in a .txt file. @smlng @aabadie would you be OK with that?

**Comment by jia200x** [comment]
thanks!

**Pull request merged**